test/automation/src/sql/managePackagesDialog.ts

```diff
@@ -60,4 +61,41 @@ export class ManagePackagesDialog extends Dialog {
 
 		return packageVersion;
 	}
+
+	async removePackage(packageName: string, clickOnTab: boolean = false): Promise<void> {
+		if (clickOnTab) {
+			const installedPkgTab = `${ManagePackagesDialog.dialogPage} div[class="tab-header"][aria-controls="dialogPane.Manage Packages.0"]`;
+			await this.code.waitAndClick(installedPkgTab);
+		}
+
+		// Wait for initial loading spinner to disappear
+		const loadingSpinner = `${ManagePackagesDialog.dialogPage} div.modelview-loadingComponent-spinner`;
+		await this.code.waitForElement(loadingSpinner);
+		await this.code.waitForElementGone(loadingSpinner);
+
+		// Click on package row in installed packages list to select it for uninstall
+		const packageRow = `${ManagePackagesDialog.dialogPage} div[role="gridcell"][aria-label="${packageName}"]`;
+		await this.code.waitAndClick(packageRow);
+
+		// Tab over to uninstall button on the right side of the row. Can't select the uninstall button
```

Comment:
This seems extremely fragile. Doesn't the button have some sort of aria label or something we can use to select it?

Reply:
We can select the button, but we can't correlate it to the package row we're looking for. Each row in the installed packages slickgrid consists of [ name, version, UninstallButton ], and only the name column has an aria-label that matches the package name. I tried using the page's uninstall button below the grid, but very frequently that button isn't visible when launching the smoke tests, so clicking on it seems to fail. ADS keeps launching at half size in my local testing, so you have to scroll to get to the bottom of the page to click on the button.

Reply:
![GridRows](https://user-images.githubusercontent.com/12820011/176289007-982969ab-f089-439b-9336-9d54c26063b7.png)

That Trash can icon is what this test is tabbing over to.